### PR TITLE
Fix: Allow users to update image with error

### PR DESCRIPTION
**Pull request by biancarosa**
# Description

This pull-requests adds the ability to update a failed version of an image, as the user might change something on the next version and/or retry without changing anything due to internal server errors.

Fixes #THEEDGE-1278

## Type of change

What is it?

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation update
- [ ] Tests update

# Checklist:

- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I run `go fmt ./...` to check that my code is properly formatted
- [x] I run `go vet ./...` to check that my code is free of common Go style mistakes

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -48,7 +48,10 @@ test:
 	go test $$(go list ./... | grep -Ev '(/test/|/ownershipvoucher|/fdo)') $(TEST_OPTIONS)
 
 coverage: 
-	go test $$(go list ./... | grep -v /test/) $(TEST_OPTIONS) -coverprofile=coverage.txt -covermode=atomic
+	go test $$(go list ./... | grep -Ev '(/test/|/ownershipvoucher|/fdo)') $(TEST_OPTIONS) -coverprofile=coverage.txt -covermode=atomic
```
Comment (Avielyo10) [inline Makefile:51]: in our Github action we test fdo && ownershipvoucer, for local testing you can use `$(command -v podman || command -v docker) run --rm -ti -v $(pwd):/edge-api quay.io/cloudservices/edge-api:libfdo-data
`

**Review by acosferreira — APPROVED** [review]
LGTM

File: Makefile (hunk 1 above)
Comment (biancarosa) [reply Makefile]: Hey @Avielyo10, I undid this change as it's completely unrelated to my PR :) Tks for the info!

**Review by Avielyo10 — APPROVED** [review]
lgtm

**Pull request merged**